mne/io/base.py

```diff
@@ -684,63 +685,36 @@ def set_annotations(self, annotations, emit_warning=True):
-                offset = (new_annot.orig_time - meas_date -
-                          self.first_samp / self.info['sfreq'])
-            else:
-                offset = 0
-
-            omit_ind = list()
-            omitted = limited = 0
-            for ind, onset in enumerate(new_annot.onset):
-                onset += offset
-                if onset > self.times[-1]:
-                    omitted += 1
-                    omit_ind.append(ind)
-                    logger.debug('Omitting %d @ %s > %s'
-                                 % (ind, onset, self.times[-1]))
-                elif onset < self.times[0]:
-                    if onset + new_annot.duration[ind] < self.times[0]:
-                        omitted += 1
-                        omit_ind.append(ind)
-                        logger.debug('Omitting %d @ %s < %s'
-                                     % (ind, onset, self.times[0]))
-                    else:
-                        limited += 1
-                        duration = new_annot.duration[ind] + onset
-                        new_annot.duration[ind] = duration
-                        new_annot.onset[ind] = self.times[0] - offset
-                elif (onset + new_annot.duration[ind] >
-                      self.times[-1] + 1.1 / self.info['sfreq']):
-                    # We have to permit onset+duration to appear to go one past
-                    # the last sample in order to actually include the last
-                    # sample...
-                    limited += 1
-                    new_annot.duration[ind] = (self.times[-1] +
-                                               1. / self.info['sfreq'] -
-                                               onset)
-            new_annot.onset = np.delete(new_annot.onset, omit_ind)
-            new_annot.duration = np.delete(new_annot.duration, omit_ind)
-            new_annot.description = np.delete(new_annot.description, omit_ind)
-            if emit_warning:
-                if omitted > 0:
-                    warn('Omitted %s annotation(s) that were outside data '
-                         'range.' % omitted)
-                if limited > 0:
-                    warn('Limited %s annotation(s) that were expanding '
-                         'outside the data range.' % limited)
-
-        self._annotations = new_annot
+        if annotations is None:
+            self._annotations = None
+        else:
+            _ensure_annotation_object(annotations)
+            time_of_frist_sample = (
+                _handle_meas_date(self.info['meas_date']) +
+                (self.first_samp / self.info['sfreq']))
+            delta = 1. / self.info['sfreq']
+
+            new_annotations = annotations.copy()
+            if annotations.orig_time is None:
+                # Asume annotations to be relative to the data
+                new_annotations.orig_time = time_of_frist_sample
+
+            tmin = time_of_frist_sample
+            tmax = tmin + self.times[-1] + delta
```

Comment:
I'm not sure that we should add `delta`. To me it seems that I've been sloppy with the `>` and `>=` within crop.